src/3d/qgsvirtualpointcloudentity_p.cpp

```diff
@@ -130,9 +130,18 @@ void QgsVirtualPointCloudEntity::handleSceneUpdate( const SceneContext &sceneCon
   const QVector<QgsPointCloudSubIndex> subIndexes = provider()->subIndexes();
   for ( int i = 0; i < subIndexes.size(); ++i )
   {
+    // If the chunked entity needs an update, do it even if it's occluded,
+    // since otherwise we'd return needsUpdate() == true until it comes into
+    // view again.
+    bool needsUpdate = mChunkedEntitiesMap.contains( i ) && mChunkedEntitiesMap[i]->needsUpdate();
+
     const QgsBox3D &box3D = mBboxes.at( i );
 
-    if ( box3D.isEmpty() )
+    if ( !needsUpdate && box3D.isEmpty() )
+      continue;
+
+    QgsAABB aabb = QgsAABB::fromBox3D( box3D, mBboxesEntity->vertexDataOrigin() );
```

Comment:
Hmm I _think_ this is not correct. isCullable() needs AABB that has coordinates relative to the scene's origin, but mBboxesEntity->vertexDataOrigin() is an arbitrarily picked origin for coordinates in mBboxesEntity. I believe you have copied this conversion from QgsVirtualPointCloudEntity::getNearFarPlaneRange() where I introduced it incorrectly - I believe it should read mMapSettings->origin() instead...

For a test, maybe try changing origin of the 3D map scene so that it is not near mBboxesEntity->vertexDataOrigin() - and hopefully you will be able to see some incorrect behavior...